guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,437 @@
+# How to use Hyperlane token (TIA) as gas token in your rollup
+
+This guide builds upon [CosmWasm rollup](../tutorials/cosmwasm.md), where you set up a local rollup with integrated CosmWasm. In this guide, we will be exploring how to integrate Hyperlane as a safer IBC alterative for token transfers in your rollup.
+
+<!-- markdownlint-disable MD033 -->
+<script setup>
+import Callout from '../.vitepress/components/callout.vue'
+</script>
+
+:::tip
+<Callout />
+:::
+
+<!-- markdownlint-enable MD033 -->
+
+## 📝 Deploy a CosmWasm rollup {#deploy-rollup}
+
+This is a gist of the [CosmWasm rollup](../tutorials/cosmwasm.md) guide.
+
+### Prepare rollup binary
+
+```bash
+git clone --branch v0.50.0 --depth 1 https://github.com/CosmWasm/wasmd.git
+cd wasmd
+```
+
+#### Add token factory to app.go:
+
+```bash
+curl -sSL https://raw.githubusercontent.com/Stride-Labs/Rollkit-docs/hyperlane-v2/guides/assets/hyperlane-use-tia-for-gas/add-tokenfactory.diff | git apply
+```
+
+#### Update packages to use rollkit and accommodate x/tokenfactory:
+
+```bash
+# Use rollkit
+go mod edit -replace github.com/cosmos/cosmos-sdk=github.com/rollkit/cosmos-sdk@v0.50.7-rollkit-v0.13.6-no-fraud-proofs.0.20240730125236-04ca9ba69219
+
+# Update the dependencies to accommodate x/tokenfactory
+go mod edit -replace cosmossdk.io/core=cosmossdk.io/core@v0.11.0
+go mod edit -replace github.com/cosmos/ibc-go/v8=github.com/cosmos/ibc-go/v8@v8.0.0
+go get -v github.com/Stride-Labs/tokenfactory@a301020c313d6bd8b45e474ca242922bdd133f6b
+
+# Perform module tidying and download dependencies
+go mod tidy -compat=1.17
+go mod download
+```
+
+#### Build the rollup binary locally
+
+```bash
+make install
+```
+
+### Configure Local Clients
+
+Config the `wasmd` and `strided` CLI for ease of use:
+
+```bash
+wasmd config set client chain-id localwasm
+wasmd config set client node tcp://127.0.0.1:36657
+wasmd config set client output json
+wasmd config set client keyring-backend test
+
+git clone --branch v24.0.0 --depth 1 https://github.com/Stride-Labs/stride ../stride
+cd ../stride
+make install
+strided config chain-id stride-internal-1
+strided config node https://stride-testnet-rpc.polkachu.com:443
+strided config keyring-backend test
+cd ../wasmd
+```
+
+Add the signer accounts to each keyring
+
+```bash
+echo "join always addict position jungle jeans bus govern crack huge photo purse famous live velvet virtual weekend hire cricket media dignity wait load mercy" | \
+  wasmd keys add my-key --recover
+
+echo "join always addict position jungle jeans bus govern crack huge photo purse famous live velvet virtual weekend hire cricket media dignity wait load mercy" | \
+  strided keys add my-key --recover
+```
+
+:::tip
+In this guide, we're using a predefined seed phrase for simplicity. Alternatively you can generate and use your own:
```

Comment:
Let's break this into 2. 

The first about simplicity should be above the steps for using that seed phrase. 

The second that can stay here is about generating your own.